third_party/move/move-vm/runtime/src/move_vm.rs

```diff
@@ -65,6 +70,22 @@ impl MoveVM {
         Session {
             move_vm: self,
             data_cache: TransactionDataCache::new(remote),
+            module_store: ModuleStorageAdapter::new(self.runtime.module_storage()),
+            native_extensions,
+        }
+    }
+
+    /// Create a new session, as in `new_session`, but provide native context extensions and custome storage for resolved modules.
+    pub fn new_session_with_extensions_and_modules<'r>(
```

Comment:
This is the new api added that would allow for a stateless MoveVM. 

Note that the module storage uses `Arc<dyn ModuleStorage>` instead of `&'r dyn` and I did this for a reason. I wanted to keep the public api for session creation to remain unchanged. Thus I put a module cache in `VMRuntime` and use this cache as the default storage for resolved modules. If the api takes a `&'r dyn`, we would have to make the lifetime of loader to be at least the lifetime of the `&'r`, which isn't always the case.

Reply:
Trying to understand, what is the relationship between `TransactionDataCache` and `ModuleStorageAdapter` now?

Reply:
As explained previously, `TransactionDataCache` is for getting raw serialized data from the chain and for storing side effects such as value change.  `ModuleStorageAdapter` is for storing the loaded modules. With that being said, I think it may make sense to just put this adapter into the `TransactionDataCache` in the future?